packages/react-dom/src/client/ReactDOM.js

```diff
@@ -179,6 +180,8 @@ if (exposeConcurrentModeAPIs) {
       queueExplicitHydrationTarget(target);
     }
   };
+
+  ReactDOM.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.getIsHydrating = getIsHydrating;
```

Comment:
It's very important that this isn't exposed in the CM release. It won't be needed by the future API neither.

Instead, put it behind a specific feature flag that we won't enable.